posthog/api/organization_feature_flag.py

```diff
@@ -96,10 +99,57 @@ def copy_flags(self, request, *args, **kwargs):
                 )
                 continue
 
-            context = {
-                "request": request,
-                "team_id": target_project_id,
-            }
+            # get all linked cohorts, sorted by creation order
+            seen_cohorts_cache = {}
+            sorted_cohort_ids = flag_to_copy.get_cohort_ids(
+                seen_cohorts_cache=seen_cohorts_cache, sort_by_creation_order=True
+            )
+
+            # destination cohort id is different from original cohort id - create mapping
+            name_to_dest_cohort_id = {}
+            # create cohorts in the destination project
+            if len(sorted_cohort_ids):
+                for cohort_id in sorted_cohort_ids:
+                    original_cohort = seen_cohorts_cache[str(cohort_id)]
+
+                    # search in destination project by name
+                    destination_cohort = Cohort.objects.filter(
+                        name=original_cohort.name, team_id=target_project_id, deleted=False
+                    ).first()
+
+                    # create new cohort in the destination project
+                    if not destination_cohort:
+                        new_filters = copy.deepcopy(original_cohort.filters)
+                        if new_filters:
+                            properties = new_filters.get("properties", [])
+                            for outer_prop in properties.get("values", []):
+                                for inner_prop in outer_prop.get("values", []):
+                                    if inner_prop.get("type") == "cohort":
+                                        original_child_cohort_id = str(inner_prop["value"])
+                                        original_child_cohort = seen_cohorts_cache[original_child_cohort_id]
+                                        inner_prop["value"] = name_to_dest_cohort_id[original_child_cohort.name]
+
+                        destination_cohort = Cohort.objects.create(
```

Comment:
use the CohortSerializer to generate the cohort, it handles calculating the cohort & populating all necessary fields.